### ISSUE TITLE
Default number called for favorite contacts

### ISSUE THREAD

**Issue by tb06** [SimpleMobileTools/Simple-Dialer#223]
Hello:
When I touch a standard contact, it opens its file in the contact app, so that I can choose which number to call: OK, perfect for me.
When I touch a favorite contact it directly launches a call but which number of the contact is called by default?
I think we should have an option to select the number we want to dial and the ability to set a default number if we want (it already exists in the Google app).
Thanks in advance!

**Comment by chaoscreater** [comment]
I'm having a similar problem.

If I choose a contact from the Contacts list, I can pick one of their numbers to call. If that particular contact is added to Favourites and I choose them from the Favourites menu, then it immediately tries to call them but doesn't actually dial any number. This is because it doesn't seem to know which number to call, even if you have set a default number on that contact.

Basically, calling a contact from the Favourites menu is useless. I have to find the contact in the Contacts menu and call them from there.

**Comment by Aga-C** [comment]
@chaoscreater Do you mean the popup on the screenshot below? If yes, just tap on it, and then you can choose the number to call.

![image](https://user-images.githubusercontent.com/85929121/141614258-4c040238-33ff-4470-bdad-c8d64269f0b1.png)

**Comment by chaoscreater** [comment]
> 
> 
> @chaoscreater Do you mean the popup on the screenshot below? If yes, just tap on it, and then you can choose the number to call.
> 
> ![image](https://user-images.githubusercontent.com/85929121/141614258-4c040238-33ff-4470-bdad-c8d64269f0b1.png)

Oh awesome, didn't know that! Yes that worked. But would still be nice if it would just dial using the default number straight away.

**Comment by tb06** [comment]
> @chaoscreater Do you mean the popup on the screenshot below? If yes, just tap on it, and then you can choose the number to call.
> 
> ![image](https://user-images.githubusercontent.com/85929121/141614258-4c040238-33ff-4470-bdad-c8d64269f0b1.png)

I just tried with a favorite contact I don't call often and indeed it offers me the choice to select between the 4 numbers this contact has.
But for another contact I call often (my mother!) it never asks me to choose between the landline number and the mobile one, and it calls the latter… I didn't find anywhere how to choose the default number for this contact, or the reinitialize it.

**Comment by Jumbo125** [comment]
Is a default number option for favourites avaibel?
Can't find the option. If a choose the contact in the full list, I have the label behind the number. If I choose the contact from favourite list, I didn't have the label. So it's more difficult. If there is a default number option, it clear twice problems.
Thank you for your great work!

**Comment by knut-hildebrandt** [comment]
Well, for me the behavior of the dialer app is a bit confusing. As above mentioned it changes when one changes the tab from "contacts" to "favorites" to "dialed" numbers.

With only one SIM card present it is like this:
1. contacts: 
short touch (tap) -> it tries to open contact app -  I'm actually  asked which to open
long touch -> opening contact app, deleting and sending SMS offered
2. favorites:
short touch (tap) -> all numbers of the contact are offered to choose one from
long touch -> opening contact app and sending SMS offered
3. dialed numbers:
short touch (tap) -> number is dialed directly even though it is a contact with various numbers
long touch -> adding to contacts, deleting from list, sending SMS and more options offered

With two SIM cards present and activated it is like this:
1. contacts: 
short touch (tap) -> it tries to open contact app -  I'm actually  asked which to open
long touch -> asking which SIM card to use, all other opttions (opening contact app, deleting and sending SMS) from menu
2. favorites:
short touch (tap) -> all numbers of the contact are offered to choose one from
long touch -> asking which SIM card to use, all other opttions (opening contact app, deleting and sending SMS) from menu
3. dialed numbers:
short touch (tap) -> number is dialed directly even though it is a contact with various numbers
long touch -> asking which SIM card to use, all other opttions (opening contact app, deleting and sending SMS) from menu

With two SIM cards present and only one activated it is like with only one present.

I would propose a unified approach which almost always works the same way independently if there are one or more SIM cards present or in which tab I am. It could look like this:

1. with only one active SIM card
short touch (tap) -> dial directly the (standard) number, if there are more than one ask for it and if it should be the future standard
long touch ->  offer all available numbers (as under favorites now) and ask if it it should be the future standard before dialing

2. with two active SIM cards
short touch (tap) -> same behavior as above using standard SIM card, if there is none an dialogue to choose one and maybe make it standard should be opened
long touch ->  offer all available numbers and SIM cards before dialing and also ask if they should be the future standards

And thanks for this great app that does not give away all my contact details to google as the pre-installed do.

**Comment by knut-hildebrandt** [comment]
> > @chaoscreater Do you mean the popup on the screenshot below? If yes, just tap on it, and then you can choose the number to call.
> > ![image](https://user-images.githubusercontent.com/85929121/141614258-4c040238-33ff-4470-bdad-c8d64269f0b1.png)
> 
> I just tried with a favorite contact I don't call often and indeed it offers me the choice to select between the 4 numbers this contact has. But for another contact I call often (my mother!) it never asks me to choose between the landline number and the mobile one, and it calls the latter… I didn't find anywhere how to choose the default number for this contact, or the reinitialize it.

Same here. For some contacts there seem the be set default numbers, for others not. But I have not the impression that this has to do with the frequency of calling them. Maybe it is caused by being marked as default in the pre-installed dialer?

**Comment by tibbi** [comment]
Simple Contacts already supports default numbers
